src/acl/DomainData.cc

```diff
@@ -78,24 +80,23 @@ aclDomainCompare(T const &a, T const &b)
             // When a.example.com comes after .example.com in an ACL
             // sub-domain is ignored. That is okay. Just important
             bool d3big = (strlen(d3) > strlen(d4)); // Always suggest removing the longer one.
-            debugs(28, DBG_IMPORTANT, "WARNING: '" << (d3big?d3:d4) << "' is a subdomain of '" << (d3big?d4:d3) << "'");
-            debugs(28, DBG_IMPORTANT, "WARNING: You should remove '" << (d3big?d3:d4) << "' from the ACL named '" << AclMatchedName << "'");
+            debugs(28, DBG_IMPORTANT, "WARNING: '" << (d3big?d3:d4) << "' is a subdomain of '" << (d3big?d4:d3) << "'. " <<
+                    "You should remove '" << (d3big?d3:d4) << "' from the ACL.");
             debugs(28, 2, "Ignore '" << d3 << "' to keep splay tree searching predictable");
         }
     } else if (ret == 0) {
         // It may be an exact duplicate. No problem. Just drop.
         if (strcmp(d1,d2)==0) {
-            debugs(28, 2, "WARNING: '" << d2 << "' is duplicated in the list.");
-            debugs(28, 2, "WARNING: You should remove one '" << d2 << "' from the ACL named '" << AclMatchedName << "'");
+            debugs(28, DBG_PARSE_NOTE(2), "WARNING: '" << d2 << "' is duplicated in the list. " <<
+                    "You should remove one '" << d2 << "' from the ACL.");
             return ret;
         }
         // When a.example.com comes before .example.com in an ACL
         // discarding the wildcard is critically bad.
         // or Maybe even both are wildcards. Things are very weird in those cases.
         bool d1big = (strlen(d1) > strlen(d2)); // Always suggest removing the longer one.
```

Comment:
Since we have to modify these messages, I suggest improving this code readability by introducing `longer` and `shorter` variables:

```suggestion
        bool d1big = (strlen(d1) > strlen(d2)); // Always suggest removing the longer one.
        const auto longer = d1big ? d1 : d2;
        const auto shorter = d1big ? d2 : d1;
```

Same for other similar contexts, of course.